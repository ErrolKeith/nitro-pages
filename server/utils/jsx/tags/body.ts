import { makeJsx } from "./makeJsx";

export interface BodyConfig {
  props: BodyProps;
  children: React.ReactElement[];
}

export interface BodyProps {
  className?: string;
}

export function makeBody(config: BodyConfig) {
  return makeJsx("body", config.props, config.children);
}
