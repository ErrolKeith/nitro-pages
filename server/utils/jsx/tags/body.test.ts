import { it, expect } from "vitest";
import { makeBody } from "./body";
import { jsx } from "react/jsx-runtime";

it("can make a body tag", () => {
  const body = makeBody({
    props: {},
    children: [],
  });
  const bodyWithChildren = makeBody({
    props: {},
    children: [jsx("p", { children: "test-body-content" })],
  });
  expect(body).toMatchSnapshot();
  expect(bodyWithChildren).toMatchSnapshot();
});
