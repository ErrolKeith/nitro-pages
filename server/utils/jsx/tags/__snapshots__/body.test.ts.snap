// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make a body tag 1`] = `<body />`;

exports[`can make a body tag 2`] = `
<body>
  <p>
    test-body-content
  </p>
</body>
`;
